build.mill

```diff
@@ -775,7 +776,30 @@ object dist extends MillPublishJavaModule {
     val isWin = scala.util.Properties.isWin
     val outputPath = T.dest / (if (isWin) "run.bat" else "run")
 
-    os.write(outputPath, prependShellScript())
+    val prependShellScript = {
+      val (millArgs, otherArgs) =
+        forkArgs().partition(arg => arg.startsWith("-DMILL") && !arg.startsWith("-DMILL_VERSION"))
+      // Pass Mill options via file, due to small max args limit in Windows
+      val vmOptionsFile = T.dest / "mill.properties"
+      val millOptionsContent =
+        millArgs.map(_.drop(2).replace("\\", "/")).mkString(
+          "\r\n"
+        ) // drop -D prefix, replace \ with /
+      os.write(vmOptionsFile, millOptionsContent)
+      val jvmArgs = otherArgs ++ List(
+        s"-DMILL_OPTIONS_PATH=$vmOptionsFile",
+        s"-Dmill.main.cli=$outputPath"
+      )
+      val classpath = runClasspath().map(_.path.toString)
+      launcherScript(
+        jvmArgs,
+        jvmArgs,
+        classpath,
+        Agg(pathingJar().path.toString) // TODO not working yet on Windows! see #791
+      )
+    }
+
+    os.write(outputPath, prependShellScript)
```

Comment:
What's the motivation to make this local?

Reply:
`prependShellScript` was a task only used the `launcher` task. I needed the output of `prependShellScript` to hard-code (in a `milll.main.cli` Java property) the path of the launcher generated by `launcher`. Somehow, I needed `launcher` to pass an argument to `prependShellScript` (an argument involving `T.dest`), so I initially inlined `prependShellScript`. The new push makes it stop being a task, and makes it a standard method accepting arguments instead.